master/internal/task/allocation.go

```diff
@@ -377,21 +377,6 @@ func (a *Allocation) RequestResources(ctx *actor.Context) error {
 // Cleanup ensures an allocation is properly closed. It tries to do everything before failing and
 // ensures we don't leave any resources running.
 func (a *Allocation) Cleanup(ctx *actor.Context) {
-	// These messages must be sent when the actor is closing since it
-	// closes all websockets listening for these events.
-	exitReasons := strings.Join(a.exitReasons, ", ")
-	if exitReasons != "" {
-		a.logger.Insert(ctx, a.enrichLog(model.TaskLog{
-			Log:   fmt.Sprintf("exit reasons: %s", exitReasons),
-			Level: ptrs.Ptr(model.LogLevelWarning),
-		}))
-	}
-	exitErr := okExitMessage
-	if a.exitErr != nil {
-		exitErr = a.exitErr.Error()
-	}
-	a.sendEvent(ctx, sproto.Event{ExitedEvent: &exitErr})
-
```

Comment:
I don't understand what this code is doing, or why it is safe to delete?

Reply:
I just moved this logic to the `a.terminated()` function instead.